### PR TITLE
Fix for wdio-video-reporter with multi remote wdio

### PR DESCRIPTION
**Pull request by ajitesh11**
This PR provide the fix for issue
https://github.com/webdriverio-community/wdio-video-reporter/issues/67

This has been tested for below use cases :
1. When wdio run in non multi remote mode
```
capabilities: [{
  browserName: 'chrome',
}],
```
2. When wdio run in multi remote mode with one browser (browser capability is given custom name)
```
capabilities: {
  Mybrowser1 : {
    capabilities: {
      browserName: 'chrome',
    },
  }
}
```
3. When wdio run in multi remote mode with two browser - Here we get video from one browser only. But tests do not stop and runs as they should. Now how to get two videos in one test that can be a future enhancement.
```
capabilities: {
  Mybrowser1 : {
    capabilities: {
      browserName: 'chrome',
    },
  },
  Mybrowser2 : {
    capabilities: {
      browserName: 'chrome',
    },
  }
}
```

Since this is my first contribution to open source repo, I might have missed something to add in PR description, so If there is any thing else needs to be added as part of PR description please let me know.

### CODE REVIEW

**Comment by ajitesh11** [comment]
Hi @christian-bromann , I am back on this, Last time it failed with error Jest: "global" coverage threshold for branches (98%) not met: 95.62% , I have added few more tests now and coverage looks like below:

image

could you please re triggered the test pipeline again for this one, as only maintainer can trigger it for PR coming from first time contributor.

**Comment by ajitesh11** [comment]
Hi @christian-bromann Could you please approve running workflow ?

File: src/frameworks/cucumber.js [hunk 1]
```diff
@@ -44,14 +44,19 @@ export default {
...
 
       let browserName = 'browser';
-      if(browser.capabilities.browserName) {
-        browserName = browser.capabilities.browserName.toUpperCase();
-      } else if(browser.capabilities.deviceName) {
-        browserName = `${browser.capabilities.deviceName.toUpperCase()}-${browser.capabilities.platformName.toUpperCase()}`;
+
+      const capabilities = this.isMultiremote
+      ? this.capabilities[Object.keys(this.capabilities)[0]]
+      : this.capabilities;
```
Comment (christian-bromann) [inline src/frameworks/cucumber.js:50]:
```suggestion
        ? this.capabilities[Object.keys(this.capabilities)[0]]
        : this.capabilities;
```

File: src/frameworks/cucumber.js [hunk 2]
```diff
@@ -67,11 +72,15 @@ export default {
    */
   onSuiteEnd (suite) {
     if (config.usingAllure) {
-      if (browser.capabilities.deviceType) {
-        allureReporter.addArgument('deviceType', browser.capabilities.deviceType);
+      const capabilities = this.isMultiremote
+      ? this.capabilities[Object.keys(this.capabilities)[0]]
+      : this.capabilities;
```
Comment (christian-bromann) [inline src/frameworks/cucumber.js:77]:
```suggestion
        ? this.capabilities[Object.keys(this.capabilities)[0]]
        : this.capabilities;
```

Comment (christian-bromann) [reply src/frameworks/cucumber.js:77]: Maybe we can put this assignment into a function, given we do the same above as again

File: src/index.js [hunk 3]
```diff
@@ -63,20 +68,35 @@ export default class Video extends WdioReporter {
...
+    this.isMultiremote = runner.isMultiremote || false;
+
+    const sessionId = runner.isMultiremote ?
+      Object.entries(runner.capabilities).map(([, capabilities]) => capabilities.sessionId)
+      : runner.sessionId;
+    this.sessionId = sessionId;
+
+    const runnerInstance = runner.isMultiremote ?
+      runner.instanceOptions[sessionId[0]]
+    : runner.instanceOptions[sessionId];
```
Comment (christian-bromann) [inline src/index.js:82]:
```suggestion
    const runnerInstance = runner.isMultiremote
      ? runner.instanceOptions[sessionId[0]]
      : runner.instanceOptions[sessionId];
```

File: src/index.js [hunk 4]
```diff
@@ -158,11 +178,16 @@ export default class Video extends WdioReporter {
     this.testnameStructure.pop();
 
     if(config.usingAllure) {
-      if (browser.capabilities.deviceType) {
-        allureReporter.addArgument('deviceType', browser.capabilities.deviceType);
+
+      const capabilities = this.isMultiremote
+      ? this.capabilities[Object.keys(this.capabilities)[0]]
+      : this.capabilities;
```
Comment (christian-bromann) [inline src/index.js:184]:
```suggestion
        ? this.capabilities[Object.keys(this.capabilities)[0]]
        : this.capabilities;
```

**Review by christian-bromann — COMMENTED** [review]
Thanks for pinging. Some minor edits.

File: src/frameworks/cucumber.js (hunk 2 above)
Comment (ajitesh11) [reply src/frameworks/cucumber.js]: Sure, have created a function in helpers.js and used it across. Please review it again and approve running workflow @christian-bromann 
Thanks :)

File: src/frameworks/default.js [hunk 5]
```diff
@@ -57,14 +57,17 @@ export default {
...
     let browserName = 'browser';
-    if(browser.capabilities.browserName) {
-      browserName = browser.capabilities.browserName.toUpperCase();
-    } else if(browser.capabilities.deviceName) {
-      browserName = `${browser.capabilities.deviceName.toUpperCase()}-${browser.capabilities.platformName.toUpperCase()}`;
+    const capabilities = helpers.getCurrentCapabilities(this);
+
+
+    if(capabilities.browserName) {
+    browserName = capabilities.browserName.toUpperCase();
```
Comment (christian-bromann) [inline src/frameworks/default.js:64]:
```suggestion
      browserName = capabilities.browserName.toUpperCase();
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍  Thanks!

**Comment by ajitesh11** [comment]
Thanks @christian-bromann :) If its possible, Could you please give me "write access" to this repo so that I can merge it? Username : ajitesh11 , Email : ajitesh11@gmail.com

**Comment by christian-bromann** [comment]
@ajitesh11 I can merge and release a new version. Happy to give you also write access if I see more PRs like that 😉 . We don't really have policies around giving write access to contributor of community packages. In the main repo 10 qualified PRs are required to get write access. I think in the community packages we can use a smaller number, e.g. 5.

**Comment by ajitesh11** [comment]
@christian-bromann Thank you :) I understand and surely more PR's will be incoming from me, as and when I find both opportunity and time :)

**Pull request merged**